### PR TITLE
Disable ota updates on certain android flavors

### PR DESCRIPTION
**Pull request by pazos**
Requires https://github.com/koreader/android-luajit-launcher/pull/126
Related to #2040

### CODE REVIEW

File: frontend/device/android/device.lua [hunk 1]
```diff
@@ -10,6 +10,13 @@ local T = require("ffi/util").template
 local function yes() return true end
 local function no() return false end
 
+local function canUpdateApk()
+    local flavor = android.getFlavor()
+    -- disable updates on fdroid builds, since they manage their own repo.
+    -- disable updates on debug builds too.
```
Comment (Frenzie) [inline frontend/device/android/device.lua]: Is that necessary? It seems like it just complicates testing without a clear gain.

Comment (pazos) [reply frontend/device/android/device.lua]: disable on debug builds? not necessary

Comment (Frenzie) [reply frontend/device/android/device.lua]: Well, I wouldn't then. :-)

Comment (pazos) [reply frontend/device/android/device.lua]: ok, but are you fine with the flavor of debug builds being "debug" or shall I change that too (and make the flavor settable, like on release builds)?

Comment (Frenzie) [reply frontend/device/android/device.lua]: Oh right, that should definitely be settable.

Comment (Frenzie) [reply frontend/device/android/device.lua]: Kinda defeats the point of a debug build otherwise, imho.

**Comment by pazos** [comment]
Done!

BTW, this PR breaks the remote option of reproducible builds for android (ie: resources won't match between two or more different flavors). I think this is fine as both short and long term solution.

File: frontend/device/android/device.lua [hunk 2]
```diff
@@ -21,7 +26,7 @@ local Device = Generic:new{
     display_dpi = android.lib.AConfiguration_getDensity(android.app.config),
     hasClipboard = yes,
     hasColorScreen = yes,
-    hasOTAUpdates = yes,
+    hasOTAUpdates = function() return canUpdateApk() end,
```
Comment (houqp) [inline frontend/device/android/device.lua]: wouldn't it be simpler to do the following?

```lua
hasOTAUpdates = canUpdateApk
```

**Comment by Frenzie** [comment]
> BTW, this PR breaks the remote option of reproducible builds for android (ie: resources won't match between two or more different flavors). I think this is fine as both short and long term solution.

That would mean there's no logical possibility for reproducible builds of any kind of "flavors" at all, on any platform. All that matters is randomness introduced during the build process, such as filesystem order and timestamps. This PR indirectly does that, but not in a way that is unique for the need to adjust the mtimes of generated files. Get rid of the randomness and what you're left with is reproducible, provided you use the same not at all random set of control switches.

**Pull request merged**